### ISSUE TITLE
kube-proxy can no longer print its version

### ISSUE THREAD

**Issue by fasaxc** [kubernetes/kubernetes#46026]
**Is this a request for help?** (If yes, you should use our troubleshooting guide and community support channels, see http://kubernetes.io/docs/troubleshooting/.):

No

**What keywords did you search in Kubernetes issues before filing this one?** (If you have found any duplicates, you should instead reply there.):

"kube-proxy version"

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):

BUG REPORT

**Kubernetes version** (use `kubectl version`):

1.7.0-alpha.3

**Environment**:
- **Cloud provider or hardware configuration**: n/a
- **OS** (e.g. from /etc/os-release): Ubuntu 16.04.2 LTS (Xenial Xerus)
- **Kernel** (e.g. `uname -a`): Linux smc-ubuntu 4.4.0-75-generic #96-Ubuntu SMP Thu Apr 20 09:56:33 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
- **Install tools**: N/A
- **Others**:

**What happened**:

I ran `kube-proxy --version`, got an error instead of the version:
```
kube-proxy --version
W0518 11:35:39.647049   73974 server.go:155] WARNING: all flags other than --config and --cleanup-iptables are deprecated. Please begin using a config file ASAP.
Starting CPU profile
W0518 11:35:39.685089   73974 server.go:314] Neither --kubeconfig nor --master was specified. Using default API client. This might not work.
error: no configuration has been provided
```

**What you expected to happen**:

`kube-proxy` prints its version

**How to reproduce it** (as minimally and precisely as possible):

```
<check out 1.7.0-alpha.3>
make kube-proxy
_output/bin/kube-proxy --version
```

**Anything else we need to know**:

**Comment by dcbw** [comment]
Confirmed.  Odd, since kube-proxy does include `verflag.PrintAndExitIfRequested()` but for some reason, that's not working.  Adding `pflag.CommandLine.AddFlagSet(command.Flags())` to kube-proxy's main() gets us "--help" output, but it still runs into the Complete() issue that requires a config option.  Not sure what the real answer is here.

**Comment by qiujian16** [comment]
it because kube-proxy does not include ``` pflag.Parse() ```  so --version does not work, but not sure why it is removed...

**Comment by cmluciano** [comment]
This might be related to the prior k8s.io/kubernetes/pkg/util/flag group being moved to k8s.io/apiserver. Let me look through the code.

**Comment by ncdc** [comment]
I'll fix it if that's ok? This is because of one of my PRs.

**Comment by cmluciano** [comment]
sounds good @ncdc 

/unassign @cmluciano

/assign @ncdc

**Comment by ncdc** [comment]
@qiujian16 we want to move all the server components to use `cobra.Command` instead of having them be empty implementations that are solely used to generate docs (#34732). As part of this transition, we won't be calling `pflag.Parse()` any more because Cobra does that for us. kube-proxy was the first one to switch, along with getting support for a config file instead of dozens of flags. See #34727. This was an unintended consequence of that change.

**Comment by ncdc** [comment]
Proposed fix in #46122
